aws/resource_aws_launch_configuration.go

```diff
@@ -526,11 +527,26 @@ func resourceAwsLaunchConfigurationRead(d *schema.ResourceData, meta interface{}
 	d.Set("ebs_optimized", lc.EbsOptimized)
 	d.Set("spot_price", lc.SpotPrice)
 	d.Set("enable_monitoring", lc.InstanceMonitoring.Enabled)
-	d.Set("security_groups", lc.SecurityGroups)
 	d.Set("associate_public_ip_address", lc.AssociatePublicIpAddress)
+	if len(lc.SecurityGroups) > 0 {
+		securityGroups := make([]string, 0, len(lc.SecurityGroups))
+		for _, sg := range lc.SecurityGroups {
+			securityGroups = append(securityGroups, *sg)
+		}
+		d.Set("security_groups", securityGroups)
+	}
+	if *lc.UserData != "" {
+		d.Set("user_data", userDataHashSum(*lc.UserData))
+	}
 
 	d.Set("vpc_classic_link_id", lc.ClassicLinkVPCId)
-	d.Set("vpc_classic_link_security_groups", lc.ClassicLinkVPCSecurityGroups)
+	if len(lc.ClassicLinkVPCSecurityGroups) > 0 {
```

Comment:
This can be simplified and error checked with:

```go
if err := d.Set("vpc_classic_link_security_groups", flattenStringList(lc.ClassicLinkVPCSecurityGroups)); err != nil {
  return fmt.Errorf("error setting vpc_classic_link_security_groups: %s", err)
}
```